### PR TITLE
Java - Rules "Avoid returning a JPA Entity in a RestController"

### PR DESCRIPTION
**Pull request by MP-Aubay**
Team : 2 millions

Rule ID incoming (https://github.com/green-code-initiative/ecoCode-challenge/issues/58)

**Rule description** : 
_Using persistence Entity in the return of a Rest controler is not energy efficient.
This use induces a multiples unnecessary SQL calls, multiple unused fields serialization, and thus unnecessary calculations by the CPU, RAM usage and network usage._

### CODE REVIEW

File: java-plugin/src/main/java/fr/greencodeinitiative/java/checks/AvoidReturningSpringEntityInRestController.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        List<MethodTree> methods = classTree.members()
+                .stream()
+                .filter(member -> member.is(Tree.Kind.METHOD))
+                .map(MethodTree.class::cast)
+                .filter(method -> method.symbol().isPublic())
+                .collect(Collectors.toList());
+
+        methods.stream().filter(methodTree -> {
+            var returnType = methods.get(0).returnType().symbolType();
+            var parametrizedTypes = returnType.typeArguments();
```
Comment (natixis-caen) [inline java-plugin/src/main/java/fr/greencodeinitiative/java/checks/AvoidReturningSpringEntityInRestController.java:58]: typo: parameterized is missing an e

Comment (MP-Aubay) [reply java-plugin/src/main/java/fr/greencodeinitiative/java/checks/AvoidReturningSpringEntityInRestController.java]: thx fixed

**Comment by MP-Aubay** [comment]
This feature should not be merge before final rule ID is know and updated in code

File: java-plugin/src/main/java/fr/greencodeinitiative/java/checks/AvoidReturningSpringEntityInRestController.java (hunk 1 above)
Comment (rducasse) [reply java-plugin/src/main/java/fr/greencodeinitiative/java/checks/AvoidReturningSpringEntityInRestController.java]: Also replaced the methods.get(0) by methodTree

**Comment by SebastienSurier** [comment]
For now, can we just add a ruleID with a number not already used and add the rule un rules.md file ? 
If necessary, we can redo a small refacto to change the naming, when a rule is well defined

**Comment by dedece35** [comment]
hi @glalloue, @jhertout
can we accept this PR without study reference ?
we have only some explanations without real study.

**Comment by jhertout** [comment]
Hello,

I agree that we don't have enough elements to validate and add this rule in the plugin.

**Comment by dedece35** [comment]
Hi @MP-Aubay, @natixis-caen, @SebastienSurier,

sorry but with implementation is that situation, we can't accept this implementation because : 
- no scientific paper to prove this rule
- no measures to prove it's a "green" rule

We keep this proposition to measure it when we will have a tool to measure it.
(check [RULES.md](https://github.com/green-code-initiative/ecoCode/blob/main/RULES.md#rules-to-be-reworked--measured--clarified) )

**Pull request closed without merging**